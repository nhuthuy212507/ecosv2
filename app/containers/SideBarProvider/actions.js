/*
 *
 * SideBarProvider actions
 *
 */

import {
  CHANGE_DESKTOP_SIDEBAR_VISIBILITY,
  CHANGE_MOBILE_SIDEBAR_VISIBILITY,
} from './constants';

export function changeDesktopSidebarVisibility() {
  return {
    type: CHANGE_DESKTOP_SIDEBAR_VISIBILITY,
  };
}

export function changeMobileSidebarVisibility() {
  return {
    type: CHANGE_MOBILE_SIDEBAR_VISIBILITY,
  };
}
