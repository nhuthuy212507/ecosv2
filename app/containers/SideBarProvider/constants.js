/*
 *
 * SideBarProvider constants
 *
 */

export const CHANGE_DESKTOP_SIDEBAR_VISIBILITY =
  'app/SideBarProvider/CHANGE_DESKTOP_SIDEBAR_VISIBILITY';

export const CHANGE_MOBILE_SIDEBAR_VISIBILITY =
  'app/SideBarProvider/CHANGE_MOBILE_SIDEBAR_VISIBILITY';
