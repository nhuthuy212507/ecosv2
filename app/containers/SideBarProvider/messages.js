/*
 * SideBarProvider Messages
 *
 * This contains all the text for the SideBarProvider container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.SideBarProvider';

export default defineMessages({
  dashboard: {
    id: `${scope}.dashboard`,
    defaultMessage: 'Dashboard',
  },
  customer: {
    id: `${scope}.customer`,
    defaultMessage: 'Customer',
  },
});
