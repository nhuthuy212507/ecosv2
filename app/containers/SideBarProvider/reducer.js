/*
 *
 * SideBarProvider reducer
 *
 */
import produce from 'immer';
import {
  CHANGE_DESKTOP_SIDEBAR_VISIBILITY,
  CHANGE_MOBILE_SIDEBAR_VISIBILITY,
} from './constants';

export const initialState = {
  collapse: false,
  show: false,
};

/* eslint-disable default-case, no-param-reassign */
const sideBarProviderReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case CHANGE_DESKTOP_SIDEBAR_VISIBILITY:
        draft.collapse = !state.collapse;
        break;
      case CHANGE_MOBILE_SIDEBAR_VISIBILITY:
        draft.show = !state.show;
        break;
    }
  });

export default sideBarProviderReducer;
