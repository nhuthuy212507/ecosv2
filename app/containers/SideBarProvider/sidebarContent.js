import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Icon } from 'antd';

import { SideBarContent, SideBarBlock } from 'components/SideBar';
import SideBarLink from './sidebarLink';
import messages from './messages';

export default function sidebarContent({ collapse }) {
  return (
    <SideBarContent collapse={collapse}>
      <SideBarBlock>
        <SideBarLink
          title={<FormattedMessage {...messages.dashboard} />}
          route="/"
          icon={<Icon type="area-chart" />}
        />
        <SideBarLink
          title={<FormattedMessage {...messages.customer} />}
          route="/customer"
          icon={<Icon type="team" />}
        />
      </SideBarBlock>
    </SideBarContent>
  );
}

sidebarContent.propTypes = {
  collapse: PropTypes.bool.isRequired,
};
