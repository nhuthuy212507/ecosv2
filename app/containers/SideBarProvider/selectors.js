import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the sideBarProvider state domain
 */

const selectSideBar = state => state.sidebar || initialState;

const makeSelectCollapse = () =>
  createSelector(
    selectSideBar,
    sidebarState => sidebarState.collapse,
  );

const makeSelectShow = () =>
  createSelector(
    selectSideBar,
    sidebarState => sidebarState.show,
  );

export { selectSideBar, makeSelectCollapse, makeSelectShow };
