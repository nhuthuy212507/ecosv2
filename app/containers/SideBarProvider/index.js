/**
 *
 * SideBarProvider
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { useInjectReducer } from 'utils/injectReducer';
import {
  Wrapper,
  ButtonBack,
  Scrollbar,
  SideBarDesktop,
  SideBarMobile,
} from 'components/SideBar';
import SideBarContent from './sidebarContent';
import { makeSelectCollapse, makeSelectShow } from './selectors';
import { changeMobileSidebarVisibility } from './actions';
import reducer from './reducer';

export function SideBarProvider({
  collapse,
  show,
  onChangeMobileSidebarVisibility,
}) {
  useInjectReducer({ key: 'sidebar', reducer });

  return (
    <Wrapper collapse={collapse} show={show}>
      <ButtonBack show={show} onClick={onChangeMobileSidebarVisibility} />
      <Scrollbar>
        <SideBarDesktop>
          <SideBarContent collapse={collapse} />
        </SideBarDesktop>
        <SideBarMobile>
          <SideBarContent collapse={collapse} />
        </SideBarMobile>
      </Scrollbar>
    </Wrapper>
  );
}

SideBarProvider.propTypes = {
  collapse: PropTypes.bool.isRequired,
  show: PropTypes.bool.isRequired,
  onChangeMobileSidebarVisibility: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  collapse: makeSelectCollapse(),
  show: makeSelectShow(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onChangeMobileSidebarVisibility: () =>
      dispatch(changeMobileSidebarVisibility()),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(SideBarProvider);
