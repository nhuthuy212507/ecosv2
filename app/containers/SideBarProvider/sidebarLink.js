/**
 *
 * sidebarLink
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import styled from 'styled-components';

import {
  SideBarLink,
  SideBarLinkIcon,
  SideBarLinkTitle,
} from 'components/SideBar';
import { makeSelectCollapse } from './selectors';

const StyledLink = styled(NavLink)`
  &.${'active'} {
    color: aliceblue;
  }
`;

export function sidebarLink({ collapse, title, icon, route, onClick }) {
  return (
    <StyledLink to={route} onClick={onClick} activeClassName="active">
      <SideBarLink collapse={collapse}>
        {icon ? <SideBarLinkIcon>{icon}</SideBarLinkIcon> : ''}
        <SideBarLinkTitle collapse={collapse}>{title}</SideBarLinkTitle>
      </SideBarLink>
    </StyledLink>
  );
}

sidebarLink.propTypes = {
  collapse: PropTypes.bool.isRequired,
  title: PropTypes.element.isRequired,
  route: PropTypes.string.isRequired,
  icon: PropTypes.element,
  onClick: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  collapse: makeSelectCollapse(),
});

const withConnect = connect(mapStateToProps);

export default compose(withConnect)(sidebarLink);
