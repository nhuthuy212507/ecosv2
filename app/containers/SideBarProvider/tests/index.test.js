/**
 * Test the HomePage
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter } from 'react-router-dom';

import SideBarProvider from '../index';
import configureStore from '../../../configureStore';

describe('<SideBarProvider />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <BrowserRouter>
            <SideBarProvider collapse={false} show={false} />
          </BrowserRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
