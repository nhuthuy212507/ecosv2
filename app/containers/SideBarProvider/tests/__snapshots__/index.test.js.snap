// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBarProvider /> should render and match the snapshot 1`] = `
<div
  class="Wrapper-sc-1kpiif2-0 fdqOuG"
>
  <button
    class="ButtonBack-kz9yne-0 jcVvmn"
  />
  <section
    class="Scrollbar-sc-177xb6a-0 cZXHHB"
    data-scrollbar="true"
    style="overflow: hidden; outline: none;"
    tabindex="-1"
  >
    <div
      class="scroll-content"
    >
      <div>
        <div
          class="SideBarDesktop-sc-172t9a6-0 efoupC"
        >
          <div
            class="SideBarContent-sc-14u8wks-0 dtoZRh"
          >
            <ul
              class="SideBarBlock-sc-1b6qjh3-0 fFUBet"
            >
              <a
                class="sidebarLink__StyledLink-gs9h8q-0 jnfaer"
                href="/dashboard"
              >
                <li
                  class="SideBarLink-u8n3ub-0 enzljv"
                >
                  <span
                    class="SideBarLinkIcon-fs5hl8-0 bxqexL"
                  >
                    <i
                      aria-label="icon: area-chart"
                      class="anticon anticon-area-chart"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="area-chart"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm-616-64h536c4.4 0 8-3.6 8-8V284c0-7.2-8.7-10.7-13.7-5.7L592 488.6l-125.4-124a8.03 8.03 0 0 0-11.3 0l-189 189.6a7.87 7.87 0 0 0-2.3 5.6V720c0 4.4 3.6 8 8 8z"
                        />
                      </svg>
                    </i>
                  </span>
                  <p
                    class="SideBarLinkTitle-cse4j8-0 ChVqR"
                  >
                    <span>
                      Dashboard
                    </span>
                  </p>
                </li>
              </a>
            </ul>
          </div>
        </div>
        <div
          class="SideBarMobile-sc-1jgbbp2-0 gnGJNg"
        >
          <div
            class="SideBarContent-sc-14u8wks-0 dtoZRh"
          >
            <ul
              class="SideBarBlock-sc-1b6qjh3-0 fFUBet"
            >
              <a
                class="sidebarLink__StyledLink-gs9h8q-0 jnfaer"
                href="/dashboard"
              >
                <li
                  class="SideBarLink-u8n3ub-0 enzljv"
                >
                  <span
                    class="SideBarLinkIcon-fs5hl8-0 bxqexL"
                  >
                    <i
                      aria-label="icon: area-chart"
                      class="anticon anticon-area-chart"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="area-chart"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm-616-64h536c4.4 0 8-3.6 8-8V284c0-7.2-8.7-10.7-13.7-5.7L592 488.6l-125.4-124a8.03 8.03 0 0 0-11.3 0l-189 189.6a7.87 7.87 0 0 0-2.3 5.6V720c0 4.4 3.6 8 8 8z"
                        />
                      </svg>
                    </i>
                  </span>
                  <p
                    class="SideBarLinkTitle-cse4j8-0 ChVqR"
                  >
                    <span>
                      Dashboard
                    </span>
                  </p>
                </li>
              </a>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="scrollbar-track scrollbar-track-x"
    >
      <div
        class="scrollbar-thumb scrollbar-thumb-x"
      />
    </div>
    <div
      class="scrollbar-track scrollbar-track-y"
    >
      <div
        class="scrollbar-thumb scrollbar-thumb-y"
      />
    </div>
  </section>
</div>
`;
